DataDefinitions/StarSystem.cs

```diff
@@ -367,5 +368,16 @@ private long estimateSystemValue(IList<Body> bodies)
 
             return value;
         }
+
+        public decimal? DistanceFromStarSystem(StarSystem other)
+        {
+            // Work out the distance to the other star system if we can 
+            if (x != null && y != null && z != null && other?.x != null && other.y != null && other.z != null)
```

Comment:
As above, is there merit in pushing the null-checking into `Functions.DistanceFromCoordinates()` itself?

Reply:
Ok. Update pushed.